experiments/AMIP/moist_mpi_earth/coupler_utils/bcfile_reader.jl

```diff
@@ -133,7 +132,10 @@ function update_midmonth_data!(date, bcf_info)
                 ),
             Tuple(1:length(monthly_fields)),
         )
+        bcf_info.segment_length .= FT(0)
     elseif (midmonth_idx == midmonth_idx0) && (Dates.days(date - all_dates[midmonth_idx]) < 0) # for init
+        midmonth_idx = bcf_info.segment_idx[1] -= Int(1)
```

Comment:
It looks like segment_idx and segment_idx0 are vectors, but the first element is always used. Can these be ints instead of Vectors?

The description of segment_idx0 also makes it sound like it should be a single element.

Reply:
Yeah, I wanted to make it a single element, but it needs to be mutable (at least `segment_idx` needs to be updated every month). Is there a better way of doing this? 🤔 I've heard mutable structs are not great for performance...

Reply:
good question! Maybe @jb-mackay knows? 

I think maybe the answer is that if it is mutating, it should be not stored in in the struct, but instead passed in as an argument to functions that need it.

Reply:
Hmmm, not sure exactly how you mean in this context. Would you mind giving an example?

Reply:
hey @LenkaNovak ! Happy to chat about this more. This is what I was thinking...

instead of this:
```
function foo(bc_info::BCDataInfo)
    # do something with bc_info.segment_idx[1]
end

# in main driver couper loop:
 bc_info.segment_idx[1] += 1 # or however it is updated
foo(bc_info)
```
we could instead remove `segment_idx` from the struct, and do something like this
```
function foo(bc_info::BCDataInfo, segment_idx::Int)
    # does something with segment_idx
end

# in main driver
segment_idx = 1 # initialize somehow
# in coupler loop
segment_idx += 1 # update
foo(bc_info, segment_idx)
```

Reply:
The thought was that if we have an integer valued object that is changing value every step, why store it in the struct? why not update it every step of the main coupler loop?

.Another thought along those lines - Maybe cleaner conceptualy if the BCDataInfo struct holds only static information about the BC file and how it is read and handled.